test/e2e/framework/device_plugin_util.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"io/ioutil"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"os"
+)
+
+const (
+	// SampleResourceName is the name of the example resource which is used in the e2e test
+	SampleResourceName = "fake.com/resource"
+	// SampleDevicePluginDSYAML is the path of the daemonset template of the sample device plugin. // TODO: Parametrize it by making it a feature in TestFramework.
+	SampleDevicePluginDSYAML = "test/e2e/testing-manifests/sample-device-plugin.yaml"
+)
+
+var (
+	appsScheme = runtime.NewScheme()
+	appsCodecs = serializer.NewCodecFactory(appsScheme)
+)
+
+// NumberOfSampleResources returns the number of resources advertised by a node
+func NumberOfSampleResources(node *v1.Node) int64 {
+	val, ok := node.Status.Capacity[SampleResourceName]
+
+	if !ok {
+		return 0
+	}
+
+	return val.Value()
+}
+
+// SampleDevicePlugin returns the Device Plugin pod for sample resources in e2e tests
+func SampleDevicePlugin() *v1.Pod {
+	gopath := os.Getenv("GOPATH")
+	var path string
+	if gopath == "" {
+		cwd, _ := os.Getwd()
+		// cwd is expected to be _output/usr/local/bin
+		path = cwd + "/../../../../" + SampleDevicePluginDSYAML
+	} else {
+		path = gopath + "/src/k8s.io/kubernetes/" + SampleDevicePluginDSYAML
```

Comment:
This will create an issue if you want to try running the tests from e2e generated binary, you may need to include this yaml in the generated code and read from that.